### PR TITLE
Enable customisation of playlist name

### PR DESCRIPTION
**Pull request by leccelecce**
First attempt for review. I struggled a bit with PureCSS having fewer options than I'm used to so the input is wider than ideal on large screens.

I can also only test up to the point of starting the Spotify API auth handshake as I still don't have the callback working.

image

### CODE REVIEW

File: src/routes/api.ts [hunk 1]
```diff
@@ -88,8 +95,8 @@ function setRoutes(redisClient: redis.RedisClient): express.Router {
             return res.status(500).send(getUserFromTokenResponse.error);
         }
 
-        const playlistName: string = `${sessionData.festivalDisplayName} ${sessionData.festivalYear} - Lineup List`;
-        const playlist: any = await spotifyHelper.getOrCreatePlaylist(access, user.id, playlistName);
+        const playlistName: string = sessionData.playlistName;
```
Comment (dot4qu) [inline src/routes/api.ts:98]: Can you add a null check here? If it's null or undefined default back to the same normal string used above and update the sessionData.playlistName field

File: views/personalized-lineup.handlebars [hunk 2]
```diff
@@ -1,10 +1,11 @@
 <h2 style="text-align: center;">Your personal {{festivalDisplayName}} lineup</h2>
 <div class="pure-g" style="justify-content: center; align-items: center;">
     <div class="pure-u-11-12 pure-u-lg-7-12 pure-u-md-5-12" style="text-align: center;">
```
Comment (dot4qu) [inline views/personalized-lineup.handlebars:3]: I put these backwards - the smallest fraction should be the largest screen size. Will you switch `pure-u-lg-7-12` to `pure-u-md-7-12` and `pure-u-md-5-12` to `pure-u-lg-5-12` please. That makes the wide text field below a little better

**Review by dot4qu — CHANGES_REQUESTED** [review]
I spent 20 minutes trying to get the text field to use different sizing than the whole form in the column and gave up, it's not worth it. I think the switch of responsive sizes that I mentioned should make everything narrow enough to be palatable, and I don't really care if it doesn't look that great

**Comment by dot4qu** [comment]
Also, given that you're merging more stuff (awesome and thank you!) that I haven't had time to deeply test myself, I'm switching the base branch of the repo to dev. That'll make it so you can keep merging, but I don't have to worry about deploying a whole host of changes at once if I just want to add in a lineup and deploy support of it straight away

**Comment by leccelecce** [comment]
> Also, given that you're merging more stuff (awesome and thank you!) that I haven't had time to deeply test myself, I'm switching the base branch of the repo to dev. That'll make it so you can keep merging, but I don't have to worry about deploying a whole host of changes at once if I just want to add in a lineup and deploy support of it straight away. Can you retarget this branch to `dev` when you make the above changes please

Thanks for considering the changes and taking the time to review! Re-targetted against dev now.

File: views/personalized-lineup.handlebars [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
-        <form id="generatePlaylistForm" action="/generate-spotify-playlist", method="POST">
+        <form class="pure-form" id="generatePlaylistForm" action="/generate-spotify-playlist", method="POST">
             <p style="margin-bottom: 0;">Click below to generate a Spotify playlist</p>
             <p style="font-size: smaller; margin-top: 0;">You will need to authorize Lineup List to access your Spotify. The only permissions used are for reading your existing playlists (to determine if you already have a Lineup List playlist created for this festival) and for creating a new playlist within your account.</p>
-            <input class="pure-button pure-button-primary" type="submit" value="Generate Playlist"></input>
+            <input style="text-align: center;" class="pure-input-1 pure-input-md-1-2" type="text" name="playlistName" value="{{playlistName}}" required></input>
+            <input style="margin-top: 8px;" class="pure-button pure-button-primary pure-input-1-5" type="submit" value="Generate Playlist"></input>
```
[the remark this answers is not in the document]
Comment (leccelecce) [reply views/personalized-lineup.handlebars:8]: Done

File: views/personalized-lineup.handlebars (hunk 2 above)
Comment (leccelecce) [reply views/personalized-lineup.handlebars:3]: Noted for future & done

File: src/routes/api.ts (hunk 1 above)
Comment (leccelecce) [reply src/routes/api.ts:105]: Done, but I need to do some proper testing of this (really need to sort local HTTPS certs for that) to prove that it works and I haven't broken anything, so feel free to review the code change but I'll leave this PR in draft until tested in a few days.

**Comment by leccelecce** [comment]
One thing I haven't tested is the code inside the null check path. If you'd like I can try deleting the relevant key out of Redis to check it handles correct - run out of time for today, have to get down to NOS :-)

**Comment by dot4qu** [comment]
Nah no need to test that low level, looks good. Thanks!

**Review by dot4qu — APPROVED** [review]

**Comment by dot4qu** [comment]
Made a few tweaks around the layout and wording but this and other changes merged to dev are now deployed on `master` and live! Thanks for the great contributions

**Comment by leccelecce** [comment]
Great stuff, thank you for deploying. I got a bit sidetracked but have plans for some more lineups / PRs over the next couple weeks

**Pull request merged**